File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* root, int val) {
        if(!root or(!root->next and root->val==val)) return NULL;
        if(!root->next and root->val!=val) return root;
        ListNode *head,*dummy=new ListNode(-1);
        dummy->next=root;
        head=dummy;
        while(head and head->next){
            if(head->next->val==val)
                head->next=head->next->next;
            else head=head->next;
        }
        
        return dummy->next;
    }
};

File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function removeElements(head: ListNode | null, val: number): ListNode | null {
    const dummy = new ListNode(0);
    dummy.next = head;
    let temp = dummy;

    while (temp) {
        if (temp.next?.val === val) {
            temp.next = temp.next.next;
        }
        else {
            temp = temp.next;
        }
    }

    return dummy.next;
};